logging/cloudwatch/hook.go

```diff
@@ -161,7 +161,6 @@ func (h *Hook) putBatches(flush <-chan bool, ticker <-chan time.Time) {
 
 func (h *Hook) sendBatch(batch []*cloudwatchlogs.InputLogEvent) {
 	h.m.Lock()
-	defer h.m.Unlock()
 
 	if len(batch) == 0 {
 		return
```

Comment:
I believe you will need to explicitly unlock the lock here because the deferred unlock has been removed.  Without the explicit unlock, passing in an zero length batch array could result in a deadlock.

You could also add the deferred unlock back so that you don't need to perform the explicit unlocks.